### DISCUSSION TITLE
Is there a way to use ETC2 and PVRTC on mobile?

### DISCUSSION THREAD

**Discussion by FabTsr** [DiligentGraphics/DiligentEngine#283] [Q&A]
Hello everybody,

I'm trying to learn how to use Diligent Engine and I'm experimenting with all the tutorials (which help me a lot). I'm interested in doing a project focused on mobile and was wondering if there was any possibility through interoperability or otherwise of using the PVRTC and ETC2 texture formats? 

Thank you in advance and congratulations on your wonderful product, which I've just discovered.

**Comment by TheMostDiligent** [comment]
ETC2 formats are not currently in the API, but it should not be too difficult to add them.
It is possible to create a texture using native API and attach it to Diligent object, but for unknown formats it will likely fail.

**Comment by FabTsr** [comment]
Ok thanks a lot, I've tested adding PVRTC in Vulkan locally in the engine, it works on iOS, there's just an incompatibility with MoltenVK and Metal and we have to use Morton's order to get a correct image.

**Reply by TheMostDiligent** [reply]
Hmm, wondering why shuffling has to be done manually and is not handled by the render device
